### ISSUE TITLE
Rectangle/Ellipse are not rendered when used in Checkbox template

### ISSUE THREAD

**Issue by donandren** [AvaloniaUI/Avalonia#2282]
Not all Rectangle/Ellipse are rendered properly when used in Checkbox template.

Repro steps:

1. Click CheckBox 'Check All'
2. All 10 listboxitems with checkboxes are expected to be checked, but only the first 3 are (visual/renderer problem).
3. 3 checkboxes at a time are checked, so  if you repeat 3 more time check all all checkboxes will be visually checked

Note: if we use default checkbox style all 10 will be checked, thus issue should be related to the Rectangle/Ellipse Shape (same checkbox template with ellipse shape have the same problem)

![2019-02-05_12-24-45](https://user-images.githubusercontent.com/13854631/52267235-3593c300-2941-11e9-9e16-a9cf56bbe9fe.gif)

```XAML
            <StackPanel>
                <StackPanel.Styles>
                    <Style Selector="CheckBox.custom">
                        <Setter Property="Template">
                            <ControlTemplate>
                                <Grid ColumnDefinitions="Auto,*">
                                    <Border Name="border"
                                            Background="{TemplateBinding Background}"
                                            BorderBrush="{TemplateBinding BorderBrush}"
                                            BorderThickness="{TemplateBinding BorderThickness}"
                                            Width="18"
                                            Height="18"
                                            VerticalAlignment="Center">
                                        <Panel>
                                            <Rectangle Name="checkMark"
                                                  Fill="{DynamicResource HighlightBrush}"
                                                  Width="11"
                                                  Height="10"
                                                  Stretch="Uniform"
                                                  HorizontalAlignment="Center"
                                                  VerticalAlignment="Center"
                                    />
                                            <Rectangle Name="indeterminateMark"
                                                       Fill="{DynamicResource HighlightBrush}"
                                                       Width="10"
                                                       Height="10"
                                                       Stretch="Uniform"
                                                       HorizontalAlignment="Center"
                                                       VerticalAlignment="Center"/>
                                        </Panel>
                                    </Border>
                                    <ContentPresenter Name="PART_ContentPresenter"
                                                      TextBlock.Foreground="{TemplateBinding Foreground}"
                                                      ContentTemplate="{TemplateBinding ContentTemplate}"
                                                      Content="{TemplateBinding Content}"
                                                      Margin="4,0,0,0"
                                                      VerticalAlignment="{TemplateBinding VerticalContentAlignment}"
                                                      HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
                                                      Grid.Column="1"/>
                                </Grid>
                            </ControlTemplate>
                        </Setter>
                    </Style>
                    <Style Selector="CheckBox.custom /template/ Rectangle#checkMark">
                        <Setter Property="IsVisible" Value="False"/>
                    </Style>
                    <Style Selector="CheckBox.custom:checked /template/ Rectangle#checkMark">
                        <Setter Property="IsVisible" Value="True"/>
                    </Style>
                </StackPanel.Styles>

                <CheckBox Name="selAll"  Content="Select All" />

                <ListBox Name="lb" Tag="{Binding #selAll.IsChecked}">
                    <ListBox.ItemTemplate>
                        <DataTemplate>
                            <CheckBox Classes="custom" Content="{Binding}" IsChecked="{Binding $parent[ListBox].Tag}" />
                        </DataTemplate>
                    </ListBox.ItemTemplate>
                </ListBox>
            </StackPanel>
```
code behind:
```C#
this.Get<ListBox>("lb").Items = Enumerable.Range(1, 10).Select(v => $"Item {v}").ToArray();
```

**Comment by donandren** [comment]
@grokys one way to solve this issue is just slightly modify layout manager:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Layout/LayoutManager.cs#L134-L145
if we change it to:
``` C#
        private void ExecuteArrangePass()
        {
           // remove measure condition in arrange
            while (_toArrange.Count > 0 /*&& _toMeasure.Count == 0*/)
            {
                var control = _toArrange.Dequeue();

                if (!control.IsArrangeValid && control.IsAttachedToVisualTree)
                {
                    Arrange(control);
                }
            }
        }
```

as we can see in Rectangle.cs (same in Ellipse.cs) 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Shapes/Rectangle.cs#L12
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Shapes/Rectangle.cs#L15-L19
measure is triggered after arrange and layout manager is unable to finish the layout pass, as it attempts to repeat it 3 times, but lot's of controls are not arranged, because measure is triggered during arrange

**Comment by donandren** [comment]
fixed by #2297
